### PR TITLE
[nrfconnect] Allow use of certifcates stored on the HDD for factory data

### PR DESCRIPTION
**Pull request by deveritec-jusc**
Current implementation only allows to use either
- default certificates, or
- certificates generated on demand.

This commit also add the possibility to use certificates already located on the HDD.
It makes it easier to use certificates and keys generated based on the test DCL.

### CODE REVIEW

File: config/nrfconnect/chip-module/Kconfig [hunk 1]
```diff
@@ -146,6 +146,28 @@ config CHIP_FACTORY_DATA_USE_DEFAULT_CERTS
...
+choice CHIP_FACTORY_DATA_USE_CERTS_CHOICE
+	prompt "Certificate source"
+	depends on !CHIP_FACTORY_DATA_USE_DEFAULT_CERTS
+	default CHIP_FACTORY_DATA_USE_GENERATED_CERTS
+
+	config CHIP_FACTORY_DATA_USE_GENERATED_CERTS
+		bool "generated"
+	config CHIP_FACTORY_DATA_USE_HDD_CERTS
+		bool "on HDD"
+endchoice
```
Comment (Damian-Nordic) [inline config/nrfconnect/chip-module/Kconfig:158]:
```suggestion
choice CHIP_FACTORY_DATA_CERT_SOURCE
	prompt "Attestation certificate file source"
	default CHIP_FACTORY_DATA_CERT_SOURCE_GENERATED

	config CHIP_FACTORY_DATA_CERT_SOURCE_DEVELOPMENT
		bool "Use pre-generated development certificates"
		help
	            Use pre-generated certificate files from the credentials/development/attestation/
	            directory that match the configured Product ID.
	config CHIP_FACTORY_DATA_CERT_SOURCE_GENERATED
		bool "Auto-generate certificates"
	config CHIP_FACTORY_DATA_CERT_SOURCE_USER
		bool "Use user-provided certificate files"
endchoice
```

That is, please include `CHIP_FACTORY_DATA_USE_DEFAULT_CERTS` as one of the choices. Other suggestions are just nits to make the prompts more verbose. Help texts to the remaining choices would be nice, too :)

Comment (deveritec-rosc) [reply config/nrfconnect/chip-module/Kconfig]: I renamed it and added some more help strings. I decided to stick with `CHIP_FACTORY_DATA_USE_CERTS_CHOICE` as the name for the first option to not break the configuration intentionally :)

File: config/nrfconnect/chip-module/Kconfig [hunk 2]
```diff
@@ -135,16 +135,45 @@ config CHIP_FACTORY_DATA_MERGE_WITH_FIRMWARE
...
-	default y
+# Select source of the certificates
+choice CHIP_FACTORY_DATA_CERT_SOURCE
+	prompt "Attestation certificate file source"
+	default CHIP_FACTORY_DATA_USE_DEFAULT_CERTS
+
+	config CHIP_FACTORY_DATA_USE_DEFAULT_CERTS
+		bool "Use pre-generated development certificates"
+		help
+	        Use pre-generated certificate files from the credentials/development/attestation/
```
Comment (Damian-Nordic) [inline config/nrfconnect/chip-module/Kconfig:146]: Something is wrong with indentation.

**Review by Damian-Nordic — APPROVED** [review]
Apart from the indentation issue, lgtm. Thanks!

**Review by bzbarsky-apple — APPROVED** [review]

File: config/nrfconnect/chip-module/Kconfig (hunk 2 above)
Comment (deveritec-rosc) [reply config/nrfconnect/chip-module/Kconfig]: Yeah, we are usually using space indentation, this setting clashed somehow with the Kconfig configuration. The normal hassle :)
Fixed this.

**Pull request merged**